invenio_communities/subcommunities/services/request.py

```diff
@@ -80,6 +81,82 @@ class SubCommunityRequest(RequestType):
     }
 
 
+class CreateSubcommunityInvitation(actions.CreateAndSubmitAction):
+    """Represents an accept action used to accept a subcommunity."""
+
+    def execute(self, identity, uow):
+        """Execute approve action."""
+        parent = self.request.created_by.resolve()
+        if not parent.children.allow:
+            raise ValidationError("Assigned parent is not allowed to be a parent.")
```

Comment:
minor: as discussed irl, this logic should be moved in a service method on the `SubCommunityService`.